controllers/openstackdataplanenode_controller.go

```diff
@@ -233,40 +233,47 @@ func (r *OpenStackDataPlaneNodeReconciler) Reconcile(ctx context.Context, req ct
 			return result, nil
 		}
 
-		var novaExternalCompute *novav1beta1.NovaExternalCompute
-		result, novaExternalCompute, err = deployment.DeployNovaExternalCompute(
-			ctx,
-			helper,
-			instance,
-			instance,
-			ansibleSSHPrivateKeySecret,
-			nodeConfigMap,
-			&instance.Status,
-			instance.Spec.NetworkAttachments,
-			instance.Spec.OpenStackAnsibleEERunnerImage)
-		if err != nil {
-			return result, err
-		}
+		if instance.Spec.Node.Nova != nil {
+			var novaExternalCompute *novav1beta1.NovaExternalCompute
+			result, novaExternalCompute, err = deployment.DeployNovaExternalCompute(
+				ctx,
+				helper,
+				instance,
+				instance,
+				ansibleSSHPrivateKeySecret,
+				nodeConfigMap,
+				&instance.Status,
+				instance.Spec.NetworkAttachments,
+				instance.Spec.OpenStackAnsibleEERunnerImage,
+				// There is anther nova template accessible in
+				// instanceRole.Spec.NodeTemplate.Nova
+				// Which one we should use here? Inheritance?
+				instance.Spec.Node.Nova,
```

Comment:
this operates by the instance instead of its role, so looks correct to me. @slagle WDYT?

Reply:
This needs to be rebased, and it would pick up the change where I've refactored all the deployment code into deployment.go, so that it can be used from both the node and role controllers. This includes the NovaExternalCompute handling.

Reply:
Yepp. It is on my TODO list to get this rebased and adapt to the changes on the main.

Reply:
It is rebased now.